torch/distributed/optim/functional_adam.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Dict, Optional, Tuple
+import torch
+import torch.optim.functional as F
+
+from torch import Tensor
+
+# Define a TorchScript compatible Functional Adam Optimizer
+# where we use these optimizer in a functional way.
+# Instead of using the `param.grad` when updating parameters,
+# we explicitly let the user pass gradients to the `step` function
```

Comment:
Just to clarify, so according to the comments below "user" is the DistributedOptimizer API, not RPC application user right? The call to optimizer should remain the same for RPC user?

Reply:
Yes that's right, DistributedOptimizer API actually pass those grads to the step function, let me update the comment to clarify